frontend/src/scenes/paths/pathsLogic.js

```diff
@@ -25,14 +25,6 @@ export const pathOptionsToProperty = {
     [`${CUSTOM_EVENT}`]: 'custom_event',
 }
 
-function checkRoot(nodeToVerify, paths, start) {
```

Comment:
I remember needing to add this in order to filter out paths that didn't start from a common source but were displayed as if they do which got confusing. If this is taken care of in the new query logic has it been resolved in the original logic or deemed a non-issue?

Reply:
Hi, I'm not sure about this function. In general I think this kind of filtering should be done server side, as the posthog frontend might not be the only user of the API.

In any case, I think it's not needed for the clickhouse paths, but can you explain what do you mean by "common source" here?

Reply:
Agreed this shouldve been placed serverside. I might be forgetting but if I recall correctly, because of how the filtering was done on the SQL, there are some paths that actually just start at "2_..." -> "3__". So the frontend would display that particular datapoint as if the "2_" event was the beginning element alongside the other paths

Reply:
The problem was with using a fuzzy match to determine the start point in postgres. The query worked fine, just that if you set e.g. "http://localhost:8000/" as the start point, it would return every url from that domain as the start point. This caused all that noise which the `checkRoot` function then eliminated.